src/protocol/reveal.rs

```diff
@@ -36,6 +40,27 @@ pub async fn reveal<F: Field>(
     Ok(input.left() + input.right() + share)
 }
 
+/// Given a vector containing secret shares of a permutation, this returns a revealed permutation.
+/// This executes `reveal` protocol on each row of the vector and then constructs a `Permutation` object
+/// from the revealed rows.
+#[allow(clippy::cast_possible_truncation, clippy::module_name_repetitions)]
+pub async fn reveal_a_permutation<F: Field>(
+    ctx: ProtocolContext<'_, F>,
+    permutations: &mut [Replicated<F>],
```

Comment:
Why is this plural? Are there more than one permutation? It appears to be a list of secret shared numbers, which I assume are positions from 1..n.